many: enforce quiet boot when requested

alfonsosanchezbeato opened the pull request:
Avoid printing anything from snap-bootstrap if "quiet" is present in the kernel command line. This prevents flickering that might happen otherwise in the splash screen.

pedronis commented on cmd/snap-bootstrap/main.go:
```diff
@@ -39,13 +39,6 @@ such as initramfs.
 	commandBuilders []func(*flags.Parser)
 )
 
-func init() {
-	err := logger.SimpleSetup()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "WARNING: failed to activate logging: %s\n", err)
-	}
-}
-
```
why remove this? I think it was there because some init code itself can log?

pedronis commented on osutil/kcmdline.go:
```diff
@@ -193,18 +189,40 @@ func KernelCommandLineKeyValues(keys ...string) (map[string]string, error) {
-				res := strings.SplitN(param, "=", 2)
-				// we have confirmed key= prefix, thus len(res)
-				// is always 2
+			res := strings.SplitN(param, "=", 2)
```
this can go in the outer loop no?

pedronis commented on osutil/kcmdline.go:
```diff
@@ -175,11 +175,7 @@ func KernelCommandLineSplit(s string) (out []string, err error) {
-// missing from the kernel command line or it has no value (eg. quiet), the key
-// is omitted from the returned map.
-func KernelCommandLineKeyValues(keys ...string) (map[string]string, error) {
+func kernelCommandLineKeyValues(omitNoVal bool, keys ...string) (map[string]string, error) {
```
as we are touching this, does it make sense to move it to use ParseKernelCommandline instead of *Split ?

pedronis commented on osutil/kcmdline.go:
```diff
@@ -193,18 +189,40 @@ func KernelCommandLineKeyValues(keys ...string) (map[string]string, error) {
+// values set for them in the kernel command line (eg. panic=-1). If the key is
+// missing from the kernel command line, it is is omitted from the returned map,
+// but it is added if present even if it has no value.
+func KernelCommandLineKeyValuesAll(keys ...string) (map[string]string, error) {
```
s/KeyValuesAll/AllParams/ ?

pedronis reviewed (commented):
did a pass, some questions

alfonsosanchezbeato reviewed (commented):
@pedronis comments addressed now, thanks for the review

alfonsosanchezbeato commented on boot/cmdline_test.go:
```diff
@@ -66,7 +66,7 @@ func (s *kernelCommandLineSuite) TestModeAndLabel(c *C) {
 		err   string
 	}{{
-		cmd:   "snapd_recovery_mode= snapd_recovery_system=this-is-a-label other-option=foo",
+		cmd:   "snapd_recovery_mode=install snapd_recovery_system=this-is-a-label other-option=foo",
```
Yeah, +1 on that, probably it can be used everywhere and the other function removed (see my previous comment)

alfonsosanchezbeato commented:
I have changed now the behavior of `KernelCommandLineKeyValues()` so it uses `ParseKernelCommandline()` and returns keys even when values are empty. I have rebased the changes to avoid the back and forth, as I hope it will be easier to review that way.

pedronis reviewed (approved):
thank you

mvo5 commented on logger/logger_test.go:
```diff
@@ -90,6 +90,50 @@ func (s *LogSuite) TestDefault(c *C) {
 }
 
+func (s *LogSuite) TestBootSetup(c *C) {
+	// env shenanigans
```
nice one!

mvo5 reviewed (approved):
Thank you
